pgmpy/estimators/ConstraintBasedEstimator.py

```diff
@@ -0,0 +1,648 @@
+        ... model = BayesianModel([('A', 'C'), ('B', 'C'), ('B', 'D'), ('C', 'E')])
+        >>> skel, sep_sets = ConstraintBasedEstimator.build_skeleton(model.nodes(), model.get_independencies())
+        >>> print(skel.edges())
+        [('A', 'C'), ('B', 'C'), ('B', 'D'), ('C', 'E')]
+        """
+
+        nodes = list(nodes)
+
+        if isinstance(independencies, Independencies):
+            def is_independent(X, Y, Zs):
+                return IndependenceAssertion(X, Y, Zs) in independencies
+        elif callable(independencies):
+            is_independent = independencies
+        else:
+            raise ValueError("'independencies' must be either Independencies-instance " +
+                             "or a ternary function that decides independencies.")
+
+        graph = UndirectedGraph(combinations(nodes, 2))
+        lim_neighbors = 0
+        seperating_sets = dict()
+        while not all([len(graph.neighbors(node)) < lim_neighbors for node in nodes]):
+            for node in nodes:
+                for neighbor in graph.neighbors(node):
+                    # search if there is a set of neighbors (of size lim_neighbors)
+                    # that makes X and Y independent:
+                    for seperating_set in combinations(set(graph.neighbors(node)) - set([neighbor]), lim_neighbors):
+                        if is_independent(node, neighbor, seperating_set):
+                            seperating_sets[frozenset((node, neighbor))] = seperating_set
+                            graph.remove_edge(node, neighbor)
+                            break
+            lim_neighbors += 1
+
+        return graph, seperating_sets
+
+    def test_conditional_independence(self, X, Y, Zs=[]):
+        """Chi-square conditional independence test.
+        Tests if X is independent from Y given Zs in the data.
+
+        This is done by comparing the observed frequencies with the expected
+        frequencies if X,Y were conditionally independent, using a chisquare
+        deviance statistic. The expected frequencies given independence are
+        `P(X,Y,Zs) = P(X|Zs)*P(Y|Zs)*P(Zs)`. The latter term can be computed
+        as `P(X,Zs)*P(Y,Zs)/P(Zs).
+
+        Parameters
+        ----------
+        X: int, string, hashable object
+            A variable name contained in the data set
+        Y: int, string, hashable object
+            A variable name contained in the data set, different from X
+        Zs: list of variable names
+            A list of variable names contained in the data set, different from X and Y.
+            This is the seperating set that (potentially) makes X and Y independent.
+            Default: []
+
+        Returns
+        -------
+        p_value: float
+            A significance level for the hypothesis that X and Y are dependent
+            given Zs. The p_value is the probability of falsely rejecting the
+            hypothesis that the variables are conditionally dependent. A low
```

Comment:
@chrisittner I don't think this should be a defined as a significance level. p-value is the probability of obtaining the given sample statistic or an even more extreme statistic considering the null hypothesis is true. If this obtained p-value is greater than the significance_level then we fail to reject the null hypothesis. From what I understand here, the null hypothesis is that the variables are independent and the alternate hypothesis is that they are dependent. 

Please correct me if you feel otherwise. 
And I cannot understand the statement - "falsely rejecting the hypothesis".